### PR TITLE
Make Measure interface and allow StatsRecorder to create it.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: api/src/main/java/openconsensus/stats/StatsRecorder.java [hunk 1]
```diff
@@ -25,15 +26,27 @@
...
+
+  /**
+   * Returns a new builder for a {@code Measure}.
+   *
+   * @param name Name of measure, as a {@code String}. Should be a ASCII string with a length no
+   *     greater than 255 characters.
+   * @return a new builder for a {@code Measure}.
+   * @since 0.1.0
+   */
+  Measure.Builder buildMeasure(String name);
```
Comment (songy23) [inline api/src/main/java/openconsensus/stats/StatsRecorder.java]: `measureBuilder` may be a better name.

**Comment by bogdandrutu** [comment]
Description is definitely not required, also most of the time the unit is "1" is it makes things simpler to have that as default. In my opinion even the name is not required because we don't use it for anything (in the implementation we should probably rely on the instance not the name).

**Comment by songy23** [comment]
> Description is definitely not required, also most of the time the unit is "1" is it makes things simpler to have that as default. 

Make sense to me

File: api/src/main/java/openconsensus/stats/Measurement.java [hunk 2]
```diff
@@ -18,8 +18,6 @@
 
 import com.google.auto.value.AutoValue;
 import javax.annotation.concurrent.Immutable;
-import openconsensus.stats.Measure.MeasureDouble;
-import openconsensus.stats.Measure.MeasureLong;
 
 /**
  * Immutable representation of a Measurement.
```
Comment (dinooliva) [inline api/src/main/java/openconsensus/stats/Measurement.java]: I like Measure and creating Measurement's from them but it's unclear to me what the utility of this class is. Certainly you need Measurement but the creation of Measurements should be done by Measure.

File: api/src/main/java/openconsensus/stats/StatsRecorder.java [hunk 3]
```diff
@@ -54,5 +67,5 @@
    *     measurements are associated with.
    * @since 0.1.0
    */
-  public abstract void record(List<Measurement> measurements, TagMap tags, SpanContext spanContext);
+  void record(List<Measurement> measurements, TagMap tags, SpanContext spanContext);
```
Comment (dinooliva) [inline api/src/main/java/openconsensus/stats/StatsRecorder.java]: I have some concerns about adding a dependency on Tracing from Stats - I know that it's at the API level only so it's less of a concern but is seems like you could get cyclic dependencies if you're not careful.

File: api/src/main/java/openconsensus/stats/StatsRecorder.java (hunk 1 above)
Comment (dinooliva) [reply api/src/main/java/openconsensus/stats/StatsRecorder.java]: Agreed.

**Comment by bogdandrutu** [comment]
To make progress I will merge this and post a new PR to address the remaining comments.

**Pull request merged**